shared/rangeanalysis/codeql/rangeanalysis/RangeAnalysis.qll

```diff
@@ -296,11 +373,24 @@ signature module LangSig<Semantic Sem, DeltaSig D> {
 }
 
 signature module BoundSig<LocationSig Location, Semantic Sem, DeltaSig D> {
+  /**
+   * A semantic bound, which defines a constraint on the possible values of an
+   * expression.
+   */
   class SemBound {
+    /**
+     * Gets a string representation of the semantic bound.
+     */
     string toString();
 
+    /**
+     * Gets the location of the semantic bound.
+     */
     Location getLocation();
 
+    /**
+     * Gets the expression associated with the semantic bound, given a delta.
```

Comment:
```suggestion
     * Gets an expression that equals this bound plus `delta`.
     *
     * For the zero-bound this gets integer constants equal to `delta`, and for
     * other bounds this gets expressions that are equal to the bound while `delta = 0`.
```

Reply:
I've accepted this, but I want to make it clearer that `delta` is an 'output'.  If I've understood correctly, a bound for `foo() + 1` might be the expression `call to foo()` with delta `-1`.  You can't just evaluate `getExpr(99)` and expect to materialize an expression for `foo() + 100`.

Reply:
That's right. This predicate really is (one of) the base cases of range analysis where you basically start with a range fact saying `E <= B + d` by virtue of `E = B + d` holding trivially.

Reply:
I'm not sure how to precisely say that `delta` is an "output", but I've added the example.  I think that makes it a little clearer.